controllers/components/datasciencepipelines/datasciencepipelines_controller.go

```diff
@@ -19,40 +19,68 @@ package datasciencepipelines
 import (
 	"context"
 
-	"k8s.io/apimachinery/pkg/runtime"
+	securityv1 "github.com/openshift/api/security/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 
 	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/render"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/updatestatus"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates/resources"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/reconciler"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
 )
 
-// DataSciencePipelinesReconciler reconciles a DataSciencePipelines object.
-type DataSciencePipelinesReconciler struct {
-	client.Client
-	Scheme *runtime.Scheme
-}
+var (
+	DefaultPath = odhdeploy.DefaultManifestPath + "/" + componentsv1.DataSciencePipelinesComponentName + "/base"
+)
 
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the DataSciencePipelines object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
-func (r *DataSciencePipelinesReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
-
-	// TODO(user): your logic here
-
-	return ctrl.Result{}, nil
-}
+func NewComponentReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	_, err := reconciler.ComponentReconcilerFor[*componentsv1.DataSciencePipelines](
```

Comment:
if you have rebased, then you should remove the generic type

Reply:
Yep I got that fixed, but since rebase I have been having issues with the model registry component, this is what the operator is saying:

```
{"level":"error","ts":"2024-11-05T16:33:51Z","logger":"opendatahub.controllers.DataScienceCluster","msg":"Failed to reconcile component: model-registry-operator","error":"unable to pactch object &{map[apiVersion:components.opendatahub.io/v1 kind:ModelRegistry metadata:map[annotations:map[component.opendatahub.io/management-state:Unknown] creationTimestamp:<nil> name:default-model-registry ownerReferences:[map[apiVersion:datasciencecluster.opendatahub.io/v1 blockOwnerDeletion:%!s(bool=true) controller:%!s(bool=true) kind:DataScienceCluster name:default-dsc uid:27dca5b7-aef3-4caa-8a72-95b64326b840]]] spec:map[registriesNamespace:odh-model-registries]]}: failed to create typed patch object (/default-model-registry; components.opendatahub.io/v1, Kind=ModelRegistry): .spec.registriesNamespace: field not declared in schema"
```

Reply:
seems fixed now

Reply:
After some testing I found that the CRDs I had installed were out of sync and just doing a undeploy/deploy fixed the issue with model registry